packages/react/src/hooks/useResolvedKPITokens.ts

```diff
@@ -0,0 +1,77 @@
+import { Fetcher, ResolvedKPIToken } from "@carrot-kpi/sdk";
+import { type Address, usePublicClient } from "wagmi";
+import { useEffect, useState } from "react";
+import { useIPFSGatewayURL } from "./useIPFSGatewayURL";
+import { usePreferDecentralization } from "./usePreferDecentralization";
+
+interface ResolvedKPITokensParams {
+    blacklisted?: Address[];
+}
+
+export function useResolvedKPITokens(params?: ResolvedKPITokensParams): {
+    loading: boolean;
+    resolvedKPITokens: ResolvedKPIToken[];
+} {
+    const publicClient = usePublicClient();
+    const preferDecentralization = usePreferDecentralization();
+    const ipfsGatewayURL = useIPFSGatewayURL();
+    const [resolvedKPITokens, setResolvedKPITokens] = useState<
+        ResolvedKPIToken[]
+    >([]);
+
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        let cancelled = false;
+        async function fetchData(): Promise<void> {
+            if (!cancelled) setLoading(true);
+
+            try {
```

Comment:
Let's use a `finally` here to set loading to false, it's just safer and gets always executed if anything happens.

Reply:
Fixed in ba94fdd708c0c95ea31e879ddda10c31848ce4d8.